geodatasets/tests/test_lib.py

```diff
@@ -134,3 +134,24 @@ def test_query_name():
 
     with pytest.raises(ValueError, match="No matching item found"):
         data.query_name("i don't exist")
+
+
+def test_filter_geometry():
+    options = [
+        "PoLyGoN",
+        "PoLy.GoN",
+        "POLY GON",
+        "poly-gon",
+        "Poly/Gon",
+        "poly_gon",
+        "POLY/gon",
+    ]
+
+    for option in options:
+        filtered = data.filter_by_geometry(option)
+        filtered_values = list(filtered.values())
+        for filtered_value in filtered_values:
```

Comment:
```suggestion
        assert len(filtered_values) > 0 and len(filtered_values) < len(data.flatten())
        for filtered_value in filtered_values:
```
Ensure that the filter actually returns a subset.